### PR TITLE
FX2024 Domain & Capabilities update / Optimize / Architecting for cloud

### PR DESCRIPTION
**Pull request by arthurclares**
This PR adds the new architecting for cloud capability that is part of the optimize domain according to the new FinOps framework.

This doc was created by Pedro Sousa (pesousa)

### CODE REVIEW

**Comment by arthurclares** [comment]
@all-contributors add @pesousa for documentation

**Comment by arthurclares** [comment]
@all-contributors add @pedrocsousa for documentation

File: docs/_docs/framework/capabilities/optimize/architecting.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+- [🍎 Learn more at the FinOps Foundation](#-learn-more-at-the-finops-foundation)
+- [⏩ Next steps](#-next-steps)
+- [🧰 Related tools](#-related-tools)
+
+</details>
+
+---
+
+<a name="definition"></a>
+**The Architecting for Cloud capability involves designing and implementing cloud infrastructure and applications in a manner that optimizes cost, performance, scalability, and reliability while aligning with business objectives.**
```
Comment (flanakin) [inline docs/_docs/framework/capabilities/optimize/architecting.md:36]:
```suggestion
**Architecting for the cloud involves designing and implementing cloud infrastructure and applications in a manner that optimizes cost, performance, scalability, and reliability while aligning with business objectives.**
```

File: docs/_docs/framework/capabilities/optimize/architecting.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+{: .fs-6 .fw-300 }
+
+This capability encompasses the architectural decisions and best practices aimed at maximizing the value derived from cloud investments while minimizing unnecessary expenditure.
+
+<br>
+
+## ✋ Before you begin
+
```
Comment (flanakin) [inline docs/_docs/framework/capabilities/optimize/architecting.md:44]:
```suggestion
## ✋ Before you begin
```

File: docs/_docs/framework/capabilities/optimize/architecting.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+Integrating the [Cloud Adoption Framework (CAF)](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/)and the [Well-Architected Framework (WAF)](https://learn.microsoft.com/azure/well-architected/pillars) with the FinOps approach is crucial for a comprehensive and effective cloud governance strategy, especially when leveraging Microsoft Azure. Here's a refined list of prerequisites considering these frameworks:
```
Comment (flanakin) [inline docs/_docs/framework/capabilities/optimize/architecting.md:46]:
```suggestion
Integrating the [Cloud Adoption Framework (CAF)](https://learn.microsoft.com/azure/cloud-adoption-framework) and the [Well-Architected Framework (WAF)](https://learn.microsoft.com/azure/well-architected/pillars) with the FinOps approach is crucial for a comprehensive and effective cloud governance strategy, especially when leveraging Microsoft Azure. Here's a refined list of prerequisites considering these frameworks:
```

File: docs/_docs/framework/capabilities/optimize/architecting.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+- **Microsoft Azure Fundamentals:** Gain familiarity with Azure services and features, including compute, storage, networking, databases, and security, aligning with the CAF's guidelines for Azure adoption.
+- **Architectural Principles and WAF Pillars:** Familiarize yourself with architectural best practices outlined in the Well-Architected Framework's pillars: operational excellence, security, reliability, performance efficiency, and cost optimization.
+- **Azure Resource Management and CAF Landing Zones:** Learn how to manage Azure resources using Azure Resource Manager (ARM) templates or Infrastructure as Code (IaC) tools like Azure Bicep or Terraform. Understand the concept of CAF landing zones for implementing Azure environments aligned with best practices.
+
```
Comment (flanakin) [inline docs/_docs/framework/capabilities/optimize/architecting.md:52]:
```suggestion
- **Microsoft Azure fundamentals:** Gain familiarity with Azure services and features, including compute, storage, networking, databases, and security, aligning with the CAF's guidelines for Azure adoption.
- **Architectural principles and WAF pillars:** Familiarize yourself with architectural best practices outlined in the Well-Architected Framework's pillars: operational excellence, security, reliability, performance efficiency, and cost optimization.
- **Azure Resource Management and CAF Landing Zones:** Learn how to manage Azure resources using Azure Resource Manager (ARM) templates or Infrastructure as Code (IaC) tools like Azure Bicep or Terraform. Understand the concept of CAF landing zones for implementing Azure environments aligned with best practices.
```

File: docs/_docs/framework/capabilities/optimize/architecting.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+This capability is a part of the FinOps Framework by the FinOps Foundation, a non-profit organization dedicated to advancing cloud cost management and optimization. For more information about FinOps, including useful playbooks, training and certification programs, and more, see the [Architecting for cloud](https://www.finops.org/framework/capabilities/architecting-for-cloud/) article in the FinOps Framework documentation.
+
+<br>
+
+## ⏩ Next steps
+
+Related FinOps capabilities:
+
+- [Commitment Discounts](docs/_docs/framework/capabilities/optimize/commitment-discounts.md)
```
Comment (flanakin) [inline docs/_docs/framework/capabilities/optimize/architecting.md:99]:
```suggestion
- [Workload optimization](docs/_docs/framework/capabilities/optimize/workloads.md)
- [Rate optimization](docs/_docs/framework/capabilities/optimize/rates.md)
```

**Review by flanakin — APPROVED** [review]

**Pull request merged**